### PR TITLE
Add reconnect automatically enhancement

### PR DESCRIPTION
**Pull request by 808sAndBR**
## Description

Adding the enhancement requested in #989 to automatically reconnect to the database when the server closes the connection.

I couldn't figure out how to add testing for this since I couldn't find a way to force a disconnect to trigger the auto reconnection. 

## Checklist

- [x] I've added this contribution to the `changelog.rst`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

**Comment by j-bennet** [comment]
Brilliant! `mycli` already has this feature, and it's great. I wanted to add it to `pgcli` for months!

There's no need for the extra config flag. It's not like the extra prompt ("Connection reset. Reconnect?") is a user-friendly feature. Let's just attempt to automatically reconnect by default.

Thank you for the PR!

**Comment by 808sAndBR** [comment]
Awesome! I made some changes to set this as the default/only behavior... If this isn't quite what you meant though, let me know and I'm happy to adjust however you would like it.

I'm excited to contribute! I love this tool, thanks for maintaining such clear contributor documentation and issue comments so that it was easy to jump in!

File: pgcli/main.py [hunk 1]
```diff
@@ -812,15 +812,14 @@ def _evaluate_command(self, text):
...
-            show_default=False, type=bool, default=True)
-        if reconnect:
-            try:
-                self.pgexecute.connect()
-                click.secho('Reconnected!\nTry the command again.', fg='green')
-            except OperationalError as e:
-                click.secho(str(e), err=True, fg='red')
+        try:
+            self.pgexecute.connect()
+            click.secho('Auto Reconnected!', fg='green')
```
Comment (j-bennet) [inline pgcli/main.py]: I suggest saying "Reconnected!" here (no "Auto").

File: pgcli/main.py [hunk 2]
```diff
@@ -812,15 +812,14 @@ def _evaluate_command(self, text):
...
+            text = self.get_last_query()
+            self.execute_command(text)
+        except OperationalError as e:
+            click.secho('Auto Reconnect Failed', fg='red')
```
Comment (j-bennet) [inline pgcli/main.py]: Also, no need for "Auto".

File: pgcli/main.py [hunk 3]
```diff
@@ -812,15 +812,14 @@ def _evaluate_command(self, text):
...
-            'Connection reset. Reconnect (Y/n)',
-            show_default=False, type=bool, default=True)
-        if reconnect:
-            try:
-                self.pgexecute.connect()
-                click.secho('Reconnected!\nTry the command again.', fg='green')
-            except OperationalError as e:
-                click.secho(str(e), err=True, fg='red')
+        try:
+            self.pgexecute.connect()
```
Comment (j-bennet) [inline pgcli/main.py]: Perhaps add a "Reconnecting..." message before the attempt here.

**Comment by 808sAndBR** [comment]
Thanks for the great feedback and for the instructions on how to force the disconnect (much easier than just waiting an hour or so whenever I make changes). 

I put in all the output text updates you requested, all sound great! I'm also now passing `text` from the query that was just attempted to the retry opposed to getting the last successful run 😬great catch on that part.

**Comment by j-bennet** [comment]
Looks good! I'll have some time and try it out tomorrow (2/21/19).

**Comment by j-bennet** [comment]
Works great. Merging! Thank you for the PR.

🍓

**Comment by 808sAndBR** [comment]
Yay! Thanks for all your help getting this into its current state! I'm super excited to have been able to contribute back to a tool that I spend so much time using 🙌

**Comment by igorkaludjer** [comment]
Thank you @808sAndBR and @j-bennet, works great! Would it be a good idea to also reconnect automatically in case of `SSL SYSCALL error: Operation timed out`?

**Pull request merged**